testsuites/syncgateway/upgrade/testserver/sg_replicate/conftest.py

```diff
@@ -0,0 +1,713 @@
+
+        if sync_gateway_version >= "2.5.0" and server_version >= "5.5.0":
+            # if SG pre-upgrade version is 2.5+, set delta sync property, otherwise, don't specify in cluster config
+            if delta_sync_enabled:
+                log_info("Running with delta sync")
+                persist_cluster_config_environment_prop(cluster_config, 'delta_sync_enabled', True)
+            else:
+                log_info("Running without delta sync")
+                persist_cluster_config_environment_prop(cluster_config, 'delta_sync_enabled', False)
+
+    if sg_ssl:
+        log_info("Enabling SSL on sync gateway")
+        persist_cluster_config_environment_prop(cluster_config, 'sync_gateway_ssl', True)
+        target1_url = "wss://{}:4984/{}".format(sg1_ip, sg_db1)
+        target1_admin_url = "wss://{}:4985/{}".format(sg1_ip, sg_db1)
+        target2_url = "wss://{}:4984/{}".format(sg3_ip, sg_db2)
+        target2_admin_url = "wss://{}:4985/{}".format(sg3_ip, sg_db2)
+    else:
+        persist_cluster_config_environment_prop(cluster_config, 'sync_gateway_ssl', False)
+
+    if cbs_ssl:
+        log_info("Running tests with cbs <-> sg ssl enabled")
+        # Enable ssl in cluster configs
+        persist_cluster_config_environment_prop(cluster_config, 'cbs_ssl_enabled', True)
+    else:
+        log_info("Running tests with cbs <-> sg ssl disabled")
+        # Disable ssl in cluster configs
+        persist_cluster_config_environment_prop(cluster_config, 'cbs_ssl_enabled', False)
+
+    if use_views:
+        log_info("Running SG tests using views")
+        # Enable sg views in cluster configs
+        persist_cluster_config_environment_prop(cluster_config, 'sg_use_views', True)
+    else:
+        log_info("Running tests not using views")
+        # Disable sg views in cluster configs
+        persist_cluster_config_environment_prop(cluster_config, 'sg_use_views', False)
+
+    persist_cluster_config_environment_prop(cluster_config, 'sg_platform', "centos", False)
+
+    # Write the number of replicas to cluster config
+    persist_cluster_config_environment_prop(cluster_config, 'number_replicas', number_replicas)
+    cluster_utils = ClusterKeywords(cluster_config)
+    cluster_topology = cluster_utils.get_cluster_topology(cluster_config)
+
+    if not skip_provisioning:
+        log_info("Installing Sync Gateway + Couchbase Server")
+
+        try:
+            cluster_utils.provision_cluster(
+                cluster_config=cluster_config,
+                server_version=server_version,
+                sync_gateway_version=sync_gateway_version,
+                sync_gateway_config=sg_config
+            )
+        except ProvisioningError:
+            logging_helper = Logging()
+            logging_helper.fetch_and_analyze_logs(cluster_config=cluster_config, test_name=request.node.name)
+            raise
+
+    # Hit this installed running services to verify the correct versions are installed
+
```

Comment:
remove this line

Reply:
this line belongs to the logic below, I removed empty line between comment and logic so that it clarifies that comment belongs to the logic